youtube_dl/extractor/lcp.py

```diff
@@ -0,0 +1,122 @@
+        skin_name = self._search_regex(embed_regex, embed_url, 'skin name', group='skin_name', fatal=False)
+
+        # Check whether the matches failed, which might be when dealing with other players (e.g., dailymotion stream)
+        if (clip_id is None) or (player_id is None) or (skin_name is None):
+            return None
+
+        return self.__extract_from_player(display_id, clip_id, player_id, skin_name)
+
+    def __extract_embed_url(self, webpage):
+        return self._search_regex(
+            r'<iframe[^>]+src=(["\'])(?P<url>.+?)\1',
+            webpage, 'embed url', group='url')
+
+    def __extract_from_player(self, display_id, clip_id, player_id, skin_name):
+        """Extracts the JSON object containing the required media info from the embedded arkena player"""
+        arkena_url = 'http://play.arkena.com/config/avp/v1/player/media/{0}/{1}/{2}/?callbackMethod=?'.format(clip_id,
+                                                                                                              skin_name,
+                                                                                                              player_id)
+        arkena_info = self._download_webpage(arkena_url, 'clip_info_' + clip_id)
+
+        arkena_info_regex = r'\?\((?P<json>.*)\);'
+        info_json = self._parse_json(self._search_regex(arkena_info_regex, arkena_info, 'json', group='json'),
+                                     display_id)
+
+        # All videos are part of a playlist, a single video is also put in a playlist
+        media_files_info = info_json.get('Playlist')
+        if media_files_info is not None:
+            media_files_info = media_files_info[0]
+        return media_files_info
+
+    def __get_thumbnails(self, media_files_info):
+        thumbnails = []
+        media_thumbnail_info = media_files_info.get('MediaInfo', {}).get('Poster')
+        if media_thumbnail_info is not None:
+            for thumbnail in media_thumbnail_info:
+                thumbnails.append({
+                    'url': thumbnail.get('Url'),
+                    'width': int_or_none(thumbnail.get('Size'))
+                })
+        return thumbnails
+
+    def __get_video_formats(self, media_files_info):
+        formats = []
+        media_files = media_files_info.get('MediaFiles')
+
+        if media_files is not None:
+            formats.extend(self.__get_mp4_video_formats(media_files))
+            self._sort_formats(formats)
+
+        return formats
+
+    def __get_mp4_video_formats(self, media_files_json):
+        formats = []
+        mp4_files_json = media_files_json.get('Mp4')
+        if mp4_files_json is not None:
+            for video_info in mp4_files_json:
+                bitrate = int_or_none(video_info.get('Bitrate'), scale=0.001) # Scale bitrate to KBit/s
+                formats.append({
+                    'url': video_info.get('Url'),
```

Comment:
If `Url` is not in `video_info` a format with `None` `url` will be added. Since `url` is critical field no such formats should be added. Same for thumbnails.